interop/shared/src/main/scala/scalaz/zio/interop/future.scala

```diff
@@ -16,6 +17,22 @@ object future {
       }
   }
 
+  implicit class FiberObjOps(private val fiberObj: Fiber.type) extends AnyVal {
+    def fromFuture[A](_ftr: => Future[A])(ec: ExecutionContext): Fiber[Throwable, A] =
+      new Fiber[Throwable, A] {
+        private lazy val ftr                                   = _ftr
+        def join: IO[Throwable, A]                             = IO.sync(Await.result(ftr, Duration.Inf))
+        def interrupt0(ts: List[Throwable]): IO[Nothing, Unit] = join.void.asInstanceOf[IO[Nothing, Unit]]
+        def onComplete(f: ExitResult[Throwable, A] => IO[Nothing, Unit]): IO[Nothing, Unit] =
+          IO.sync {
+            ftr.onComplete {
```

Comment:
Can `Future.onComplete` ever fail? Also, are there any circumstances in which `Future.onComplete` will _not_ invoke the specified callback, _other_ than infinite futures?

Reply:
It looks like there are at least two cases where this could happen:

1. If original `Future` has fatal error; in this case it will be "reported to the ExecutionContext". In testing in Ammonite, throwing a `NoSuchMethodException` in a Future causes it to bubble out synchronously.
2. If something is thrown inside the callback.

It seems like we might want to use `IO.syncThrowable` here as well, so I went ahead and made the change; let me know if this is not a good idea.

Reply:
Looks good. 👍